iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -75,6 +78,38 @@ void addGPUMatmulSimtPassPipeline(OpPassManager &pm) {
   pm.addNestedPass<FuncOp>(createLLVMGPUPipeliningPass());
 }
 
+void addGPUMatmulTensorCorePassPipeline(OpPassManager &pm) {
+  //===--------------------------------------------------------------------===//
+  // Initial clean up.
+  //===--------------------------------------------------------------------===//
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+
+  // Distribute linalg onto warps within the workgroup.
+  pm.addNestedPass<FuncOp>(
```

Comment:
Maybe this pass needs to be split into three.
1) The current tile + distribute without lowering to loops
2) The warp distribution
3) The lowering to loops

Reply:
I do think it should be split. It should be one part reduction tiling + promotion, then tile and distribute to warp or threads.
Not sure what you mean by lowering to loops. We don't really have any lowering to loops in this path?
I was considering doing this change here but the PR is already big

Reply:
I was using lowering to loops as a catch all to lowering to scalar + vector code. Basically last step before just NVVM translation.
Fine to do it as a next step.